noir-projects/aztec-nr/aztec/src/context/avm_context.nr

```diff
@@ -128,8 +129,21 @@ impl PublicContextInterface for AvmContext {
         self.accumulate_unencrypted_logs(event_selector, log);
     }
 
-    fn consume_l1_to_l2_message(&mut self, content: Field, secret: Field, sender: EthAddress) {
-        assert(false, "'consume_l1_to_l2_message' not implemented!");
+    fn consume_l1_to_l2_message(&mut self, content: Field, secret: Field, sender: EthAddress, leaf_index: Field) {
+        let secret_hash = compute_secret_hash(secret);
+        let message_hash = compute_message_hash(
+            sender,
+            self.chain_id(),
+            /*recipient=*/self.this_address(),
+            self.version(),
+            content,
+            secret_hash
+        );
+
+        assert(self.l1_to_l2_msg_exists(message_hash, leaf_index), "Tried to consume inexistent L1-to-L2 message");
```

Comment:
Minor nitpick. I think "inexistent" is a rarely used word and "nonexistent" is more common/proper.